base-template.sh

```diff
@@ -567,52 +689,81 @@ execute_shared_hooks() {
     # split on comma and newline
     IFS="$IFS_COMMA_NEWLINE"
 
-    # Fail if the shared root is not available (if enabled)
-    FAIL_ON_NOT_EXISTING=$(git config --get githooks.failOnNonExistingSharedHooks)
-
     for SHARED_REPO in $SHARED_REPOS_LIST; do
         unset IFS
 
         set_shared_root "$SHARED_REPO"
 
-        if [ ! -f "$SHARED_ROOT/.git/config" ]; then
-            echo "! Failed to execute shared hooks in $SHARED_REPO" >&2
-            echo "  It is not available. To fix, run:" >&2
-            echo "    \$ git hooks shared update" >&2
+        if echo "$EXECUTED_SHARED_HOOKS" | grep "$SHARED_ROOT"; then
+            echo "! Warning: Shared hooks in (type: \`$SHARED_HOOKS_TYPE\`):" >&2
```

Comment:
What if you have the same shared hook in the `.shared` file and you also happen to set it in the global config?
For example you contribute to a repo with a shared hook repo and you like the hooks in that shared repo and would want to use with all your other local git repos?
Maybe we should not display this warning if we are skipping duplicate execution anyway (would be good to ensure we have tests on this), or just reduce the warning to a single line so it's not too obtrusive.
What do you think?

Reply:
Jeah you are right.
Something like:
```shell
echo "! Warning: Shared hooks (type: \`$SHARED_HOOKS_TYPE\`) in:" >&2
echo "  \`$SHARED_REPO\` already executed -> skipping..." >&2
```

Reply:
Added a test